packages/app/src/cli/models/app/loader.ts

```diff
@@ -255,34 +255,45 @@ class AppLoader {
 
       const configuration = await this.parseConfigurationFile(specification.schema, configurationPath)
 
-      const entrySourceFilePath = (
-        await Promise.all(
-          ['index']
-            .flatMap((name) => [`${name}.js`, `${name}.jsx`, `${name}.ts`, `${name}.tsx`])
-            .flatMap((fileName) => [`src/${fileName}`, `${fileName}`])
-            .map((relativePath) => path.join(directory, relativePath))
-            .map(async (sourcePath) => ((await file.exists(sourcePath)) ? sourcePath : undefined)),
-        )
-      ).find((sourcePath) => sourcePath !== undefined)
-      if (!entrySourceFilePath) {
-        this.abortOrReport(
-          output.content`Couldn't find an index.{js,jsx,ts,tsx} file in the directories ${output.token.path(
+      let entryPath
+      if (specification.singleEntryPath) {
+        entryPath = (
+          await Promise.all(
+            ['index']
+              .flatMap((name) => [`${name}.js`, `${name}.jsx`, `${name}.ts`, `${name}.tsx`])
+              .flatMap((fileName) => [`src/${fileName}`, `${fileName}`])
+              .map((relativePath) => path.join(directory, relativePath))
+              .map(async (sourcePath) => ((await file.exists(sourcePath)) ? sourcePath : undefined)),
+          )
+        ).find((sourcePath) => sourcePath !== undefined)
+        if (!entryPath) {
+          this.abortOrReport(
```

Comment:
Follow on:

* Should this validation logic be moved into the ExtensionInstance? Then it's consistent with UIExtension
* Should we be testing this?

Reply:
Issue here: https://github.com/Shopify/cli/issues/816